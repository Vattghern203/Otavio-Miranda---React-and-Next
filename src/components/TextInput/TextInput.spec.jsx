import { render, screen } from '@testing-library/react'

import { TextInput } from '.'
import userEvent from '@testing-library/user-event'

describe('<TextInput />', () => {

    const fn = jest.fn()

    it('should match the placeholder', () => {
        
        render(<TextInput handleChange={fn} searchValue={'test'} />)

        const input = screen.getByPlaceholderText(/search here/i)

        expect(input).toBeInTheDocument()
    
        expect(input.value).toBe('test')
    })

    it('should call handleChange function on each key pressed', () => {
        
        render(<TextInput handleChange={fn} />)

        const input = screen.getByPlaceholderText(/search here/i)

        const value = 'test'

        userEvent.type(input, value)

        expect(input.value).toBe(value)
        expect(fn).toHaveBeenCalledTimes(value.length)

    })

    it('should match the snapshot', () => {
        const {container} = render(<TextInput handleChange={fn} searchValue={'test'}/>)

        expect(container).toMatchSnapshot()
    })
})