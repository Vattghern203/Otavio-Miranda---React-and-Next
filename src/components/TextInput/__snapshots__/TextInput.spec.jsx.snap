// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TextInput /> should match the snapshot 1`] = `
<div>
  <input
    class="text-input"
    placeholder="Search here..."
    type="search"
    value="test"
  />
</div>
`;
